### ISSUE TITLE
mtalog and Exim

### ISSUE THREAD

**Issue by stefaweb** [mailwatch/1.2.0#173]
mtalog database is empty with Exim. Seems to work only with Sendmail and Postfix.
Not sure to understand how it works.
Where to look to implement this?

The last 3 columns are empty in rep_total_mail_by_date.php.

![capture d ecran 2015-06-18 a 10 41 15](https://cloud.githubusercontent.com/assets/12907102/8227380/12525f5c-15a7-11e5-8259-8bce73c3462a.png)

**Comment by PieterJ** [comment]
For postfix this the mtalog is filled by parsing the mail.log file. This is done in postfix_relay.php.
It is contains a few errors. I have rewritten the file and extended to log more details. (Greylisted,SPF,MTA,Noqueue) If you want I can post the file here.

**Comment by endelwar** [comment]
@PieterJ yes please, fork the repository and create a pull request as explained [here](http://docs.mailwatch.org/dev/contributing.html)

**Comment by PieterJ** [comment]
Will do this afternoon, Please note that I have also changed the rep_total_mail_by_date.php to show more columns. I don't think it's wise to merge it into mailwatch.

**Comment by endelwar** [comment]
If the new columns can be useful to someone we could add a flag (`SHOW_MORE_INFO_ON_REPORT_GRAPH` maybe?) and hide or show them based on conf.php

**Comment by PieterJ** [comment]
Will try to fork this afternoon.

**Comment by stefaweb** [comment]
Is it mail.log or maillog?

On Debian, we only have

/var/log/mail.log
/var/log/mail.info
/var/log/mail.err

**Comment by stefaweb** [comment]
And /var/log/mail.warn

**Comment by PieterJ** [comment]
On postfix/ubuntu it's /var/log/mail.log

**Comment by endelwar** [comment]
It depends on distribution and version: I remember that on red hat 9 it was maillog, on debian 6, 7 and 8 it is mail.log

**Comment by endelwar** [comment]
@PieterJ did you manged to look at this?

**Comment by stefaweb** [comment]
Can we do something for Exim to fill up the last 3 columns?

Seems to be in mailq.php but not easy to add.

**Comment by endelwar** [comment]
Exim needs more love.
I'm going to install a test exim dev machine, but this will not be fast...

**Comment by stefaweb** [comment]
Not urgent. 1.3...

Have taken a fast look inside mailq.php regs and Co. Hot!

I can help to find the right stuff to look for inside the Exim logs.

**Comment by pdwalker** [comment]
@PieterJ This one has been bothering me for a while, and I'd love to have the fix

Could you either post a diff of your changes, or email me the changed files directly?  I've love to see what changes you've made.

**Comment by stefaweb** [comment]
Fixed columns view in #342.

**Comment by endelwar** [comment]
> Can we do something for Exim to fill up the last 3 columns?

an Exim log parser should be written to mimic postfix log parsing behaviour, so that there information are reported in database
